### PR TITLE
feat(feedback): improvements and fixes for feedback container

### PR DESCRIPTION
**Pull request by priyanshu891**
Closes #

Updated feedback component which now able to visualise the feedback given on text.

#### Changelog

- Visualisation of feedback
- Fixed bug related to feedback flag

**Changed**

- Feedback component is standalone component with separate package.

**Removed**

- N/A

#### Testing / Reviewing

{{ Add descriptions, steps or a checklist for how reviewers can verify this PR works or not }}

### CODE REVIEW

**Comment by priyanshu891** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by priyanshu891** [comment]
recheck

**Comment by priyanshu891** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by priyanshu891** [comment]
recheck

**Comment by annawen1** [comment]
I see the Netlify build for the deploy preview is failing due to these errors
Screenshot 2024-02-27 at 9 39 22 AM

File: packages/chat/index.ts [hunk 1]
```diff
@@ -8,4 +8,4 @@
  */
 
 import './components/test-input/test-input.js';
-import './components/feedback/feedback.js';
+import '../feedback/components/feedback/feedback.js';
```
Comment (annawen1) [inline packages/chat/index.ts:11]: We can probably remove this line as `feedback` is it's own package and not part of the `chat` package

File: packages/feedback/components/feedback/__stories__/feedback.stories.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    ...defaultArgs,
+    /**
+     *
+     * @param {CustomEvent} event custom event fires when new message is generated
+     */ 'on-feedback-delete': (event) => console.log(event.detail),
+    /**
+     *
+     * @param {CustomEvent} event custom event fires when feedback saved
+     */
+    'on-feedback-save': (event) => console.log(event.detail),
```
Comment (annawen1) [inline packages/feedback/components/feedback/__stories__/feedback.stories.js:191]: aren't the custom events already part of `defaultArgs`?

File: packages/feedback/components/feedback/src/feedback.ts [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+   */
+  @state()
+  private formData: FeedbackData = {
+    feedback_id: '',
+    generation_id: '',
+    start_index: 0,
+    end_index: 0,
+    selected_text: '',
+    suggested_text: '',
+    feedback_type: [],
```
Comment (annawen1) [inline packages/feedback/components/feedback/src/feedback.ts:97]: Can we stick with camelCasing of variables?

File: packages/feedback/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  "publishConfig": {
+    "access": "public",
+    "provenance": true
+  },
+  "description": "Carbon for AI - feedback component",
+  "license": "Apache-2.0",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/carbon-design-system/carbon-for-ai",
+    "directory": "packages/chat"
```
Comment (annawen1) [inline packages/feedback/package.json:13]:
```suggestion
    "directory": "packages/feedback"
```

File: .gitignore [hunk 5]
```diff
@@ -28,3 +28,4 @@ custom-elements.json
 *.sln
 *.sw?
 .nx/*
+agentic.stories.js
```
[the remark this answers is not in the document]
Comment (priyanshu891) [reply .gitignore:31]: Yes, daniel needs this story for our internal demo. that's why it is added to .gitignore

File: packages/feedback/components/feedback/__stories__/feedback.stories.js [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+  },
+
+  /**
+   * Renders the template for Storybook
+   * @param {object} args Storybook arguments
+   * @returns {TemplateResult<1>}
+   */
+  render: (args) =>
+    html`
+      <!-- <h2>${args.input}</h2> -->
```
[the remark this answers is not in the document]
Comment (priyanshu891) [reply packages/feedback/components/feedback/__stories__/feedback.stories.js:164]: Not needed.

File: packages/chat/index.ts (hunk 1 above)
Comment (priyanshu891) [reply packages/chat/index.ts:11]: Removed

File: packages/feedback/components/feedback/__stories__/feedback.stories.js (hunk 2 above)
Comment (priyanshu891) [reply packages/feedback/components/feedback/__stories__/feedback.stories.js:191]: In defaultArgs I am logging whole event. But here i am logging the actual data. If there is any issue i can remove that.

File: packages/feedback/components/feedback/src/feedback.ts (hunk 3 above)
Comment (priyanshu891) [reply packages/feedback/components/feedback/src/feedback.ts]: Done

File: packages/feedback/components/feedback/src/feedback.scss [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+ */
+
+$css--plex: true !default;
+
+@use '../../../../../globals/scss/vars' as *;
+@use '@carbon/styles/scss/theme' as *;
+
+:host(#{$c4ai-prefix}-feedback) hr {
+  border: 0;
+  background: var(--cds-border-subtle);
```
Comment (annawen1) [inline packages/feedback/components/feedback/src/feedback.scss:15]: You should be able to access the carbon tokens directly

`background: $border-subtle;`

File: packages/feedback/components/feedback/src/feedback.scss [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+  block-size: 1px;
+  margin-block: 1rem;
+}
+
+:host(#{$c4ai-prefix}-feedback) cds-slug .secondary {
+  color: var(--cds-text-secondary, #525252);
```
Comment (annawen1) [inline packages/feedback/components/feedback/src/feedback.scss:21]: you can use the carbon tokens directly

`color: $text-secondary;`

File: .gitignore (hunk 5 above)
[the remark this answers is not in the document]
Comment (annawen1) [reply .gitignore:31]: got it

File: packages/feedback/components/feedback/__stories__/feedback.stories.js [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+import '../feedback';
+import Flag24 from '@carbon/web-components/es/icons/flag/24.js';
+import { html } from 'lit';
+/**
+ * More on how to set up stories at: https://storybook.js.org/docs/web-components/writing-stories/introduction
+ */
+
+// const feedbackApi = FeedbackApi.getInstance();
```
Comment (annawen1) [inline packages/feedback/components/feedback/__stories__/feedback.stories.js:16]: can we remove this?

File: packages/feedback/components/feedback/__stories__/feedback.stories.js [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+  },
+
+  /**
+   * Renders the template for Storybook
+   * @param {object} args Storybook arguments
+   * @returns {TemplateResult<1>}
+   */
+  render: (args) =>
+    html`
+      <!-- <h2>${args.input}</h2> -->
```
Comment (annawen1) [inline packages/feedback/components/feedback/__stories__/feedback.stories.js:206]: can we remove this?

File: packages/feedback/components/feedback/src/feedback.scss [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+:host(#{$c4ai-prefix}-feedback) cds-slug p {
+  margin: 0.25rem 0;
+  font-size: var(--cds-body-compact-01-font-size, 0.875rem);
```
Comment (annawen1) [inline packages/feedback/components/feedback/src/feedback.scss:26]: You should be able to use the carbon type token here as well

```
@use '@carbon/styles/scss/type' as *;

:host(#{$c4ai-prefix}-feedback) cds-slug p {
  margin: 0.25rem 0;

  @include type-style('body-compact-01');
}
```

File: packages/feedback/components/feedback/src/feedback.scss [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+  font-weight: var(--cds-body-compact-01-font-weight, 400);
+  letter-spacing: var(--cds-body-compact-01-letter-spacing, 0.16px);
+  line-height: var(--cds-body-compact-01-line-height, 1.2857);
```
Comment (annawen1) [inline packages/feedback/components/feedback/src/feedback.scss:29]: You should be able to use the carbon type token here as well

```
@use '@carbon/styles/scss/type' as *;

:host(#{$c4ai-prefix}-feedback) cds-slug p {
  margin: 0.25rem 0;

  @include type-style('body-compact-01');
}
```

File: packages/feedback/components/feedback/src/feedback.template.ts [hunk 13]
```diff
@@ -0,0 +1,272 @@
...
+ */
+function showSlug(customElementClass) {
+  const { _input: input, _model_id: model } = customElementClass;
+  return html`
+    <cds-slug size="2xs" alignment="bottom-left">
+      <div slot="body-text">
+        <!-- <p class="secondary">AI Explained</p> -->
+        <!-- <h1>84%</h1>
+        <p class="secondary bold">Confidence score</p> -->
+        <!-- <hr /> -->
```
Comment (annawen1) [inline packages/feedback/components/feedback/src/feedback.template.ts:192]: Can we remove any commented out code?

File: packages/feedback/components/feedback/src/feedback.template.ts [hunk 14]
```diff
@@ -0,0 +1,272 @@
...
+              <p class="secondary">Model type</p>
+              <p class="bold">${model}</p>`
+          : null}
+        <hr />
+        <p class="secondary">
+          To provide feedback on a specific part of a paragraph, simply select
+          the portion of text you want to comment on.
+        </p>
+      </div>
+      <!-- <cds-slug-action-button>Submit</cds-slug-action-button> -->
```
Comment (annawen1) [inline packages/feedback/components/feedback/src/feedback.template.ts:206]: Can this be removed?

File: packages/feedback/components/feedback/src/feedback.template.ts [hunk 15]
```diff
@@ -0,0 +1,272 @@
...
+    highlighted,
+    calculateTextPosition,
+  } = customElementClass;
+  return html`
+    <div
+      style="display:flex; justify-content: space-between; align-items: center; padding: 0.5rem 0;position:relative">
+      <div class="slug">
+        ${feedbackList.length !== 0
+          ? html`<span class="notify"></span>`
+          : null}${showSlug(customElementClass)}
```
[the remark this answers is not in the document]
Comment (annawen1) [reply packages/feedback/components/feedback/src/feedback.template.ts:254]: The authoring would just look like this:
```
<c4ai-feedback>
    <cds-slug slot='slug'>
    ....
    </cds-slug>
   ${Flag24({ slot: 'icon' })} ${args.output}
</c4ai-feedback>
```

File: packages/feedback/components/feedback/__stories__/feedback.stories.js (hunk 10 above)
Comment (priyanshu891) [reply packages/feedback/components/feedback/__stories__/feedback.stories.js:206]: Done

File: packages/feedback/components/feedback/__stories__/feedback.stories.js (hunk 9 above)
Comment (priyanshu891) [reply packages/feedback/components/feedback/__stories__/feedback.stories.js:16]: Done

File: packages/feedback/components/feedback/src/feedback.template.ts (hunk 14 above)
Comment (priyanshu891) [reply packages/feedback/components/feedback/src/feedback.template.ts]: Done

File: packages/feedback/components/feedback/src/feedback.template.ts (hunk 13 above)
Comment (priyanshu891) [reply packages/feedback/components/feedback/src/feedback.template.ts]: Done

File: packages/feedback/components/feedback/src/feedback.scss (hunk 12 above)
Comment (priyanshu891) [reply packages/feedback/components/feedback/src/feedback.scss]: Done

File: packages/feedback/components/feedback/src/feedback.scss (hunk 8 above)
Comment (priyanshu891) [reply packages/feedback/components/feedback/src/feedback.scss]: Done

File: packages/feedback/components/feedback/src/feedback.scss (hunk 7 above)
Comment (priyanshu891) [reply packages/feedback/components/feedback/src/feedback.scss]: Done

**Review by annawen1 — APPROVED** [review]

**Pull request merged**